### PR TITLE
String semantics

### PR DESCRIPTION
**Pull request by Lord-Left**
Hey guys - not sure if I was sufficiently exhaustive in my semantics file, let me know if you'd like me to add more content before merging!

### CODE REVIEW

File: docs/semantics/string.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+The string type encapsulates sequences of characters. Strings are declared via the `str` keyword. 
+
+## Syntax
+
+A string type declaration in general takes the form of `str myString = "Text";`.  
+
+## Semantics
+
+Initalized string objects are immutable. Future support for a mutable string class, in the manner of `StringBuilder` in Java, may be supported in our standard library.
```
Comment (pattersongp) [inline docs/semantics/string.md]: We should talk to today about this and how we're going to store strings

File: docs/semantics/string.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Initalized string objects are immutable. Future support for a mutable string class, in the manner of `StringBuilder` in Java, may be supported in our standard library. 
+
+### Operators
+
+For the purposes of the below documentation, `str1` is the string consisting of the character sequence "first" and `str2` is the string consisting of the character sequence "second" .
+
+* `^` allows for string concatenation, e.g: `str newString = str1 ^ str2;`. Returns a new string whose contents are "firstsecond". 
+* `[i]` returns a string of length 1 consisting of the character at index _i_, e.g. `str char4 = str1[4];`. Returns "t" (as indexes begin at 0 in FIRE).
+* `[:]` allows for string slicing, e.g: `str slicedString = str1[3:4];`. Returns "st".
```
Comment (pattersongp) [inline docs/semantics/string.md]: Add default parameters here

What does it mean to do `str[:10]`

File: docs/semantics/string.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+
+### Char
+
+Characters or the `Char` data type does not exist in FIRE; instead, individual characters are represented with strings of 0 or 1 elements.
```
Comment (pattersongp) [inline docs/semantics/string.md]: noice

File: docs/semantics/string.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+
+## Example
+
+```
+string myString = "Hello";
+string myString2 = "World";
+myString = myString + " ";
```
Comment (pattersongp) [inline docs/semantics/string.md]: I thought we were using the `^` operator?

File: docs/semantics/string.md (hunk 2 above)
Comment (Lord-Left) [reply docs/semantics/string.md]: Added default parameters!

File: docs/semantics/string.md (hunk 4 above)
Comment (Lord-Left) [reply docs/semantics/string.md]: Fixed!

File: docs/semantics/string.md (hunk 1 above)
Comment (Lord-Left) [reply docs/semantics/string.md]: Agreed; though I'm assuming we should leave discussion of how data types are implemented out of our semantics docs?

**Pull request merged**